libqtile/backend/x11/window.py

```diff
@@ -1561,6 +1552,14 @@ def floating(self, do_float):
                 self._float_height = self.height
             self._float_state = FloatStates.NOT_FLOATING
             self.group.mark_floating(self, False)
+            stack = list(self.qtile.core._root.query_tree())
+            tiled = [win.window.wid for win in self.group.tiled_windows]
+            tiled_stack = [win for win in stack if win in tiled and win != self.window.wid]
```

Comment:
Nice! We can avoid the list constructor as we're iterating over the generator right away. Small nit but it helps me follow the X11 backend code if we use `wid` when we're only talking about the window ids:
```suggestion
            stack = self.qtile.core._root.query_tree()
            tiled = [win.window.wid for win in self.group.tiled_windows]
            tiled_stack = [wid for wid in stack if wid in tiled and wid != self.window.wid]
```